src/backend/backend_ga4.py

```diff
@@ -36,15 +38,26 @@ def send(self, message: dict):
         if message is None:
             return
         try:
+            def _send_func(request_data):
+                try:
+                    request.urlopen(request_data)  # nosec
+                except Exception as err:
+                    pass  # nosec
             data = json.dumps(message).encode()
 
             if self.backend_url.lower().startswith('http'):
                 req = request.Request(self.backend_url, data=data)
             else:
                 log.info("Incorrect backend URL.")
                 return
+            process = multiprocessing.Process(target=_send_func, args=(req,))
+            process.daemon = True
+            process.start()
+
+            process.join(self.timeout)
+            if process.is_alive():
+                process.terminate()
 
-            request.urlopen(req)  # nosec
```

Comment:
why can't we use timeout arg for `urlopen`? This way we don't need to create a new process.

Reply:
Timeout in `urlopen` just doesn't work. I tried to use it, but `request.urlopen(req)` still hangs for about 15-20 sec. Probably this timeout works for file downloading, but I'm not sure.